### PR TITLE
Remove `OpCode.HASH160` and `OpCode.HASH256`

### PR DESCRIPTION
**Pull request by erikzhang**
Since `OpCode.HASH160` and `OpCode.HASH256` are combinations of different hash operations, we can provide some more basic hash algorithms and then remove the two combined instructions.

### CODE REVIEW

File: src/neo-vm/ExecutionEngine.cs [hunk 1]
```diff
@@ -61,8 +61,9 @@ public class ExecutionEngine : IDisposable
         public IInteropService Service { get; }
         public RandomAccessStack<ExecutionContext> InvocationStack { get; } = new RandomAccessStack<ExecutionContext>();
         public RandomAccessStack<StackItem> ResultStack { get; } = new RandomAccessStack<StackItem>();
-        public ExecutionContext CurrentContext => InvocationStack.Peek();
-        public byte[] EntryScriptHash { get; private set; }
+        public ExecutionContext CurrentContext => InvocationStack.Count > 0 ? InvocationStack.Peek() : null;
+        public ExecutionContext CallingContext => InvocationStack.Count > 1 ? InvocationStack.Peek(1) : null;
+        public ExecutionContext EntryContext => InvocationStack.Count > 0 ? InvocationStack.Peek(InvocationStack.Count - 1) : null;
```
Comment (shargon) [inline src/neo-vm/ExecutionEngine.cs]: We need to store Calling and EntryContext

File: tests/neo-vm.Tests/Types/InteropService.cs [hunk 2]
```diff
@@ -9,21 +9,6 @@ public bool Invoke(byte[] method, ExecutionEngine engine)
         {
             switch (Encoding.ASCII.GetString(method))
             {
-                case "System.ExecutionEngine.GetEntryScriptHash":
```
Comment (shargon) [inline tests/neo-vm.Tests/Types/InteropService.cs]: This syscall dissapear?

File: src/neo-vm/ExecutionEngine.cs (hunk 1 above)
Comment (erikzhang) [reply src/neo-vm/ExecutionEngine.cs]: We don't have `TAILCALL` any more. So we don't need to store them.

File: tests/neo-vm.Tests/Types/InteropService.cs (hunk 2 above)
Comment (erikzhang) [reply tests/neo-vm.Tests/Types/InteropService.cs]: https://github.com/neo-project/neo/blob/976659a5778a03f5aef88dcce56122eb471557f4/neo/SmartContract/StandardService.cs#L38-L40

They are in neo.

**Comment by erikzhang** [comment]
We can discuss whether to remove opcodes such as `CHECKSIG` in a separate issue. If removed, it means that users need to change their addresses when migrating from 2.x to 3.0.

**Comment by shargon** [comment]
In fact, we should warn the users to use regular wallets before the migration. @igormcoelho @vncoelho i remember that you have a repository for allow users to create complex wallets for charity

**Comment by shargon** [comment]
Even if we eliminate `TAILCALL`, I prefer to have `CallingScript` inside `ExecutionContext`, is more clear.

**Comment by erikzhang** [comment]
> Even if we eliminate `TAILCALL`, I prefer to have `CallingScript` inside `ExecutionContext`, is more clear.

Done.

**Pull request merged**